algorithms/spot_finding/finder.py

```diff
@@ -745,19 +746,41 @@ def __call__(self, experiments):
         import six.moves.cPickle as pickle
         from dxtbx.format.image import ImageBool
 
+        # Loop through all the experiments and get the unique imagesets
+        imagesets = []
+        imageset_scan = {}
+        for experiment in experiments:
+            if experiment.imageset not in imagesets:
+                imagesets.append(experiment.imageset)
+                imageset_scan[experiment.imageset] = copy.deepcopy(experiment.scan)
+            else:
+                imageset_scan[experiment.imageset] += experiment.scan
+
+        for imageset in imagesets:
+            imageset.set_scan(imageset_scan[imageset])
+
```

Comment:
I think the above lines will actually end up modifying the scan contained in the experiment since you are not deep copying here. If this what you intended to do?

Reply:
I think it does - this also bothers me, but i think in _current_ usage this is never written out or reused after spotfinding?

Reply:
It is possible to write out experiments after spot finding which contain a mask with hot pixels. I don't know if this is used in practice but in principle if the experiments are messed up then it could possible cause problems. I think we should try to ensure that the experimental models are consistent because this kind of thing can cause confusion down the line.

Reply:
The imageset scan is never written to disk - only the experiment ones. The only thing sent to disk is the imageset indices:

```
  "imageset": [
    {
      "__id__": "ImageSequence", 
      "template": "/Users/graeme/data/vmxi-grid/image_34016_master.h5", 
      "single_file_indices": [
        0, 
        1, 
        2, 
        3, 
        4, 
        5, 
        6, 
        7, 
        8, 
        9, 
        10, 
        11, 
----8<-----
        7371, 
        7372, 
        7373, 
        7374, 
        7375, 
        7376, 
        7377, 
        7378, 
        7379
      ], 
      "mask": "", 
      "gain": "", 
      "pedestal": "", 
      "dx": "", 
      "dy": "", 
      "params": {
        "dynamic_shadowing": "Auto", 
        "multi_panel": false
      }
    }
  ], 
```

I also worried about this...

Reply:
This code is now gone... 🤔